osbenchmark/worker_coordinator/runner.py

```diff
@@ -2422,8 +2427,15 @@ async def __call__(self, opensearch, params):
                     model_ids.add(id)
 
         for model_id in model_ids:
-            resp=await opensearch.transport.perform_request('POST', '/_plugins/_ml/models/' + model_id + '/_undeploy')
-            resp=await opensearch.transport.perform_request('DELETE', '/_plugins/_ml/models/' + model_id)
+            deployed = await _is_deployed(model_id)
+            if deployed:
+                await opensearch.transport.perform_request('POST', '/_plugins/_ml/models/' + model_id + '/_undeploy')
+                while deployed:
```

Comment:
let's change it to undeploy and check for undeploy to true before delete to make code readable

Reply:
Also, if undeploy is prerequisite , why don't we undeploy all model first and then delete in next step instead of undeploy and wait for it to deploy and delete. What do you think?